06-descriptive-analysis-in-srvyr.Rmd

```diff
@@ -1 +1,420 @@
+
+```{r}
+mtcars %>% 
+  summarize(mean_hp = mean(hp))
+```
+
+These verbs can be used in conjunction with `group_by()`, applying the functions on a group-by-group basis to create grouped summaries.
+
+```{r}
+mtcars %>% 
+  group_by(cyl) %>% 
+  summarize(mean_hp = mean(hp))
+```
+<!--TODO: end callout note-->
+
+### Bivariate/multivariate analysis
+
+Bivariate analysis concerns the relationship between two variables, while Multivariate analysis studies the relationship between multiple variables. The correlation is a number between -1 and +1 that denotes the strength of the linear relationship between two variables.
+
+* Examples: Relationship between academic performance scores in 1999 and 2000; relationship between age and income 
+
+The {srvyr} package does not include a function for calculating correlations for survey variables. We can use the {survey} package's `svyvar()` estimates the variance between the variables and then `cov2cor()` to get the correlation. Another option is to use the `svycor()` function from the {jtools} package.
+
+```{r}
+# Calculate correlation between API 2000 and API 1999.
+
+var <- survey::svyvar(~api00 + api99, design = dstrata)
+vmat <- as.matrix(var)
+attr(vmat, "var") <- NULL # Remove extra info
+attr(vmat, "statistic") <- NULL # Remove extra info
+cov2cor(vmat)
+
+# Alternatively
+dstrata |>
+  survey::svyvar(~api00 + api99, design = _) |>
+  (\(x) as.matrix(x))() |>
+  mutate(across(everything(), var, NULL)) |>
+  (\(x) cov2cor(x))()
+```
+
+## Deciding on descriptive analyses
+
+We choose which measures to look at depending on which are appropriate for the type of variable we are examining. Variables are classified as categorical/nominal, ordinal, and interval/ratio.
+
+* Categorical/nominal data: variables with levels or descriptions that cannot be ordered, such as region of the country (North, South, East, and West)
+* Ordinal data: variables that can be ordered, such as those from a Likert scale (strongly disagree, disagree, agree, and strongly agree)
+* Interval/ratio: variables that are counted or measured, such as age or income
+  * Within interval/ratio data are *discrete* variables, whose values are whole numbers, such as number of children, and *continuous* variables, whose values can lie anywhere on an interval, such as weight.
+  
+We should use different measures depending on whether our variables are interval/ratio, ordinal, or nominal.
+
+## Descriptive analysis using the {srvyr} package
+
+### Setup
+
+Recall from [Chapter 05](#c05) the general process for estimation with the {srvyr} package:
+
+1. Create a `tbl_svy` object using `srvyr::as_survey_design()` or `srvyr::as_survey_rep()`.
+2. Subset the data for subpopulations using `dplyr::filter()`, if needed.
+3. Specify domains of analysis using `dplyr::group_by()`, if needed.
+4. Within `srvyr::summarize()`, specify variables to calculate ,means, totals, proportions, quantiles, and more.
```

Comment:
General comment: need to introduce cascade() which is similar to summarize() but adds total rows

Need discussion on why filter needs to be done after making tbl_svy and not before.

Reply:
added: Filtering should be done after creating the `tbl_svy` object using `srvyr::as_survey_design()` or `srvyr::as_survey_rep()` because these functions incorporate the survey design information into the resulting object.

Please let me know if I am missing anything with that!